R/react.R

```diff
@@ -157,22 +157,89 @@ JS <- function(...) { # nolint
 #' @export
 triggerEvent <- function(inputId) {
   ReactData(
-    type = "input", id = inputId, argIdx = NULL
+    type = "event", id = inputId
   )
 }
 
-#' Set input
+#' @title Set input
 #'
-#' Creates a handler which can be used for `onChange` and similar props of 'React' components
-#' to set the value of a 'Shiny' input to one of the arguments passed to the handler.
+#' @description Creates a handler which can be used for `onChange` and similar
+#' props of 'React' components to set the value of a 'Shiny' input to one of
+#' the arguments passed to the handler.
+#'
+#' The argument `jsAccessor` can be empty (assumes `jsAccessor = 1`) or
+#' take one of the following types:
+#'
+#' - A valid javscript accessor string to be applied to the object
+#' (example: `jsAccessor = "[0].target.checked"`)
+#' - A valid javscript 0-based index (that unlike R it starts at 0)
+#'
+#' As an example, calling `setInput("some_index", 0)` is equivalent to
+#' `setInput("some_index", "[0]")`
 #'
 #' @param inputId 'Shiny' input ID to set the value on.
-#' @param argIdx Index of the argument to use as value.
-#' @return A `ReactData` object which can be passed as a prop to 'React' components.
+#' @param jsAccessor Index (numeric 0-based index) or accessor (javascript string) of the argument
+#' to use as value.
+#' @return A `ReactData` object which can be passed as a prop to 'React'
+#' components.
 #'
 #' @export
-setInput <- function(inputId, argIdx = 1) {
-  ReactData(
-    type = "input", id = inputId, argIdx = argIdx - 1
-  )
-}
+#' @examples
+#' setInput("some_id")
+methods::setGeneric(
+  "setInput",
+  function(inputId, jsAccessor) {
+    stop("Arguments not supported, see the documentation.")
+  }
+)
+
+#' @describeIn setInput Uses as index `jsAccessor = 0`
+#' @export
+#' @examples
+#' setInput("some_id", 1)
+methods::setMethod(
+  "setInput",
+  signature(inputId = "character", jsAccessor = "missing"),
+  function(inputId) {
+    setInput(inputId, 1)
+  }
+)
+
+#' @describeIn setInput Gets the value from index in jsAccessor
+#' @export
+#' @examples
+#' setInput("some_id", 2)
+methods::setMethod(
+  "setInput",
+  signature(inputId = "character", jsAccessor = "numeric"),
+  function(inputId, jsAccessor = 1) {
```

Comment:
Is this default value used in any way? If it is, make it a 0 :wink:
```suggestion
  function(inputId, jsAccessor) {
```